### PR TITLE
Restructure README

### PR DESCRIPTION
**Pull request by odow**
I've always found the README quite dense. We should be off-loading information to the documentation or the `jump.dev` website.

This PR reduces the amount of text, makes it more explicit that users should use the community forum to get help, and explains who should be using the latest release and development versions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,123 +3,56 @@
...
-a list of the significant breaking changes in the JuMP 0.19 release.**
-
-**Latest Release**: 0.21.5 (`release-0.21` branch)
-  * [Documentation](https://jump.dev/JuMP.jl/v0.21.5/)
+JuMP is a domain-specific modeling language for [mathematical optimization](https://en.wikipedia.org/wiki/Mathematical_optimization)
+embedded in [Julia](https://julialang.org/). You can find out more about us by 
+visiting [jump.dev](https://jump.dev).
+
+**Latest Release**: 0.21.5 (`release-0.21` branch):
+  * Use this version if you are using JuMP to solve problems.
```
Comment (mlubin) [inline README.md:11]: The difference between a release and development version is pretty clear. Doesn't need a sentence explaining, if we're looking at reducing the text in the readme.

File: README.md [hunk 2]
```diff
@@ -3,123 +3,56 @@
...
+## Need help?
 
-## Supported problem classes
+Use the [Community forum] to search for answers to previously asked questions, 
+or ask a new question. 
 
-Mathematical optimization encompasses a large variety of problem classes.
-We list below what is currently supported. See the documentation for more information.
+Before asking a question, you can help us by reading [PSA: make it easier to help you](https://discourse.julialang.org/t/psa-make-it-easier-to-help-you/14757),
+and following the suggestions. In particular, _simplify_ your problem and make
```
Comment (mlubin) [inline README.md:39]: Could drop "In particular, _simplify_ your problem and make it reproducible". The sentence could be interpreted as somewhat condescending. I'd prefer to phrase this more like: "here are the best practices for asking a question."

File: README.md [hunk 3]
```diff
@@ -3,123 +3,56 @@
 
 [![Powered by NumFOCUS](https://img.shields.io/badge/powered%20by-NumFOCUS-orange.svg?style=flat&colorA=E1523D&colorB=007D8A)](https://numfocus.org)
 
-JuMP is a domain-specific modeling language for **[mathematical optimization]**
-embedded in **[Julia]**. It currently supports a number of open-source and
-commercial solvers ([Artelys Knitro], [BARON], [Bonmin], [Cbc], [CDCS], [CDD],
-[Clp], [COSMO], [Couenne], [CPLEX], [CSDP], [ECOS], [FICO Xpress], [GLPK],
-[Gurobi], [Ipopt], [Juniper], [MOSEK], [NLopt], [OSQP], [ProxSDP], [SCIP],
-[SCS], [SDPA], [SDPT3], [SeDuMi], [Tulip]) for a variety of problem classes, including
-**[linear programming]**, **[(mixed) integer programming]**,
```
Comment (mlubin) [inline README.md:12]: If we're deleting these keywords from the README, we should add some as GitHub package topics for better searchability.

Comment (odow) [reply README.md:12]: Added keywords as Github package topics.

Comment (odow) [reply README.md:12]: image

**Review by mlubin — APPROVED** [review]

**Pull request merged**